### DISCUSSION TITLE
[FedGw] ModuleNotFoundError: No module named 'magma'

### DISCUSSION THREAD

**Discussion by robinge330** [magma/magma#6608] [Q&A]
After installing the FedGW version 1.4 the following error appears.
PLease could you guide what is the issue and how to resolve this

Apr 30 22:55:15 fb-1agw systemd[1]: magma@magmad.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 22:55:15 fb-1agw magmad[1704]: Traceback (most recent call last):
Apr 30 22:55:15 fb-1agw magmad[1704]:   File "/usr/local/bin/service_util.py", line 20, in <module>
Apr 30 22:55:15 fb-1agw magmad[1704]:     from magma.common.health.service_state_wrapper import ServiceStateWrapper
Apr 30 22:55:15 fb-1agw magmad[1704]: ModuleNotFoundError: No module named 'magma'
Apr 30 22:55:15 fb-1agw systemd[1]: magma@magmad.service: Control process exited, code=exited status=1
Apr 30 22:55:15 fb-1agw systemd[1]: magma@magmad.service: Failed with result 'exit-code'.
Apr 30 22:55:20 fb-1agw systemd[1]: magma@magmad.service: Service hold-off time over, scheduling restart.
Apr 30 22:55:20 fb-1agw systemd[1]: magma@magmad.service: Scheduled restart job, restart counter is at 10.
Apr 30 22:55:20 fb-1agw systemd[1]: Stopped Magma magmad service.
Apr 30 22:55:20 fb-1agw magmad[1714]: /usr/bin/python3: Error while finding module specification for 'magma.magmad.main' (ModuleNotFoundError: No module named 'magma')
Apr 30 22:55:20 fb-1agw systemd[1]: Started Magma magmad service.
Apr 30 22:55:20 fb-1agw systemd[1]: magma@magmad.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 22:55:21 fb-1agw magmad[1723]: Traceback (most recent call last):
Apr 30 22:55:21 fb-1agw magmad[1723]:   File "/usr/local/bin/service_util.py", line 20, in <module>
Apr 30 22:55:21 fb-1agw magmad[1723]:     from magma.common.health.service_state_wrapper import ServiceStateWrapper
Apr 30 22:55:21 fb-1agw magmad[1723]: ModuleNotFoundError: No module named 'magma'
Apr 30 22:55:21 fb-1agw systemd[1]: magma@magmad.service: Control process exited, code=exited status=1
Apr 30 22:55:21 fb-1agw systemd[1]: magma@magmad.service: Failed with result 'exit-code'.

**Comment by robinge330** [comment]
Hi,

Kindly could anyone guide what could be the issue here for the Federation GW.

Command: apt show magma
Package: magma
Version: 1.4.0-1618930512-8dcd7a4e
Priority: extra

Command: service magma@* status
● magma@magmad.service - Magma magmad service
   Loaded: loaded (/etc/systemd/system/magma@magmad.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Mon 2021-05-03 17:16:10 UTC; 3s ago
  Process: 15098 ExecStopPost=/usr/bin/env python3 /usr/local/bin/service_util.py magmad (code=exited, status=1/FAILURE)
  Process: 15086 ExecStart=/usr/bin/env python3 -m magma.magmad.main (code=exited, status=1/FAILURE)
 Main PID: 15086 (code=exited, status=1/FAILURE)

Command: journalctl -u magma@*
-- Logs begin at Fri 2021-04-30 20:10:27 UTC, end at Mon 2021-05-03 17:16:19 UTC. --
Apr 30 22:54:29 fb-1agw systemd[1]: Started Magma magmad service.
Apr 30 22:54:29 fb-1agw magmad[972]: /usr/bin/python3: Error while finding module specification for 'magma.magmad.main' (ModuleNotFoundError: No module named 'magma')
Apr 30 22:54:29 fb-1agw systemd[1]: magma@magmad.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 22:54:29 fb-1agw magmad[1083]: Traceback (most recent call last):
Apr 30 22:54:29 fb-1agw magmad[1083]:   File "/usr/local/bin/service_util.py", line 20, in <module>
Apr 30 22:54:29 fb-1agw magmad[1083]:     from magma.common.health.service_state_wrapper import ServiceStateWrapper
Apr 30 22:54:29 fb-1agw magmad[1083]: ModuleNotFoundError: No module named 'magma'
Apr 30 22:54:29 fb-1agw systemd[1]: magma@magmad.service: Control process exited, code=exited status=1
Apr 30 22:54:29 fb-1agw systemd[1]: magma@magmad.service: Failed with result 'exit-code'.
Apr 30 22:54:34 fb-1agw systemd[1]: magma@magmad.service: Service hold-off time over, scheduling restart.
Apr 30 22:54:34 fb-1agw systemd[1]: magma@magmad.service: Scheduled restart job, restart counter is at 1.
Apr 30 22:54:34 fb-1agw systemd[1]: Stopped Magma magmad service.
Apr 30 22:54:34 fb-1agw systemd[1]: Started Magma magmad service.
Apr 30 22:54:35 fb-1agw magmad[1387]: /usr/bin/python3: Error while finding module specification for 'magma.magmad.main' (ModuleNotFoundError: No module named 'magma')
Apr 30 22:54:35 fb-1agw systemd[1]: magma@magmad.service: Main process exited, code=exited, status=1/FAILURE
Apr 30 22:54:35 fb-1agw magmad[1395]: Traceback (most recent call last):

**Comment by robinge330** [comment]
This is resolved now. Thanks

**Reply by jayshankarnair** [reply]
I am

**Reply by jayshankarnair** [reply]
I am getting the same error . No module named magma . what is the issue.

.health.service_state_wrapper import ServiceStateWrapper
Jun  1 15:05:02 magma-dev magmad[23402]: ImportError: No module named 'magma'
Jun  1 15:05:02 magma-dev systemd[1]: magma@magmad.service: Control process exited, code=exited status=1
Jun  1 15:05:02 magma-dev systemd[1]: magma@magmad.service: Unit entered failed state.
Jun  1 15:05ver, scheduling restart.
Jun  1 15:05:02 magma-dev systemd[1]: Stopped Magma magmad service.
Jun  1 15:05:02 magma-dev systemd[1]: Started Magma magmad service.
Jun  1 15:05:02 magma-dev magmad[23400]: /usr/bin/python3: Error while finding module specification for 'magma.magmad.main' (ImportError: No module named 'magma')
Jun  1 15:05:02 magma-dev systemd[1]: magma@magmad.service: Main process exited, code=exited, status=1/FAILURE
Jun  1 15:05:02 magma-dev magmad[23402]: Traceback (most recent call last):
Jun  1 15:05:02 magma-dev magmad[23402]:   File "/usr/local/bin/service_util.py", line 20, in <module>
Jun  1 15:05:02 magma-dev magmad[23402]:     from magma.common:02 magma-dev systemd[1]: magma@magmad.service: Failed with result 'exit-code'.
